### ISSUE TITLE
Improper rendering

### ISSUE THREAD

**Issue by ashfurrow** [ashfurrow/Nimble-Snapshots#38]
On iOS 9, I'm getting views and buttons that aren't rendering properly. 

![screen shot 2015-11-04 at 2 17 46 pm](https://cloud.githubusercontent.com/assets/498212/10949976/94711cc0-8303-11e5-8247-6b560bfc0361.png)

Specifically, customizations to views like text and images aren't being respected. 

Calling something like 

```swift
view.drawViewHierarchyInRect(view.bounds, afterScreenUpdates: true)
expect(view).to(haveValidSnapshot())
```

Fixes the problem, but it's a hassle to do it every time. Using `usesDrawViewHierarchyInRect` fixes the issue but introduces many other problems (I don't trust UIKit's API for snapshots). 

I think adding to the `Snapshotable` extensions to call this method ourselves would be an OK fix. Thoughts?

**Comment by orta** [comment]
So, this problem also happened to me on Eigen, and occasionally crops up energy with Expecta+Snapshots. 

On those cases I also do the prefixed `drawHeir...` but it's a hack. Can you test if using the `FBSnapshotTestController`'s `usesDrawViewHierarchyInRect` as true makes a difference?

As running the draw could work every time, but it kinda feels like this is a problem at FBSnapshot level

**Comment by ashfurrow** [comment]
Yeah, so I need to clarify two different options:

1. Call `drawViewHierarchyInRect` _before_ calling down to Facebook's library.
1. Use Facebook's `usesDrawViewHierarchyInRect` to capture snapshots using `drawViewHierarchyInRect` _instead_ of `renderLayerWhatever`. 

I agree it's a problem best fixed by Facebook's library, but I don't feel like they'd accept option 1. And I _really_ dislike option 2, since it breaks _a tonne_ of our tests (other things render improperly). 

I'd like to go wth option 1 – is there any technical reason not to? I know it'd be a big change – possibly breaking – so I'd release a new major version.

**Comment by ashfurrow** [comment]
Pushed a possible solution ^

**Comment by orta** [comment]
only technical thing I can think of is that it adds an extra render step so _maybe_ something async could change, but IMO, that's alright. 

Def a major version change, this could break people's projects. However, IMO, shippable.

**Comment by ashfurrow** [comment]
Cool, will push a major update.

**Comment by anas10** [comment]
It seems that this fix isn't available anymore in the latest version (4.1.0).

**Comment by orta** [comment]
In 4.1.0 the record/check functions have a bool for setting useDrawRect: https://github.com/ashfurrow/Nimble-Snapshots/blob/master/HaveValidSnapshot.swift#L168
